### PR TITLE
NN-2721 improvements

### PR DESCRIPTION
**Pull request by JamieDraperUK**
- Fail if more than one CSWAP location is found
- Move query parameters into request body 
- Added more tests
- Made reason optional and defaulted to ADM

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderMovementsResource.java [hunk 1]
```diff
@@ -16,15 +16,7 @@
...
-import uk.gov.justice.hmpps.prison.api.model.CourtHearing;
-import uk.gov.justice.hmpps.prison.api.model.CourtHearingDateAmendment;
-import uk.gov.justice.hmpps.prison.api.model.CourtHearings;
-import uk.gov.justice.hmpps.prison.api.model.ErrorResponse;
-import uk.gov.justice.hmpps.prison.api.model.OffenderBooking;
-import uk.gov.justice.hmpps.prison.api.model.PrisonMoveCancellation;
-import uk.gov.justice.hmpps.prison.api.model.PrisonToCourtHearing;
-import uk.gov.justice.hmpps.prison.api.model.PrisonToPrisonMove;
-import uk.gov.justice.hmpps.prison.api.model.ScheduledPrisonToPrisonMove;
+import uk.gov.justice.hmpps.prison.api.model.*;
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderMovementsResource.java:19]: 😢

File: build.gradle [hunk 2]
```diff
@@ -87,6 +87,7 @@ dependencies {
     testImplementation 'com.github.tomakehurst:wiremock-standalone:2.27.1'
     testImplementation 'io.jsonwebtoken:jjwt:0.9.1'
     testImplementation 'io.swagger.parser.v3:swagger-parser-v2-converter:2.0.21'
+    testImplementation 'org.glassfish:javax.el:3.0.0'
```
Comment (Nimphal) [inline build.gradle:90]: Is this commitable? Very happy if so, but wanted to check it wasn't a side effect of git add .

File: src/main/java/uk/gov/justice/hmpps/prison/service/MovementUpdateService.java [hunk 3]
```diff
@@ -110,10 +117,16 @@ private OffenderBooking getActiveOffenderBooking(final Long bookingId) {
...
     private AgencyInternalLocation getCswapLocation(final String agency) {
-        return agencyInternalLocationRepository
+        final var cellSwapLocations = agencyInternalLocationRepository
                 .findByLocationCodeAndAgencyId("CSWAP", agency)
                 .stream()
                 .filter(AgencyInternalLocation::isCellSwap)
+                .collect(Collectors.toList());
+
+        if(cellSwapLocations.size() > 1) throw new RuntimeException("There are more than 1 CSWAP locations configured");
+
```
Comment (Nimphal) [inline src/main/java/uk/gov/justice/hmpps/prison/service/MovementUpdateService.java:127]: Nice!

File: build.gradle (hunk 2 above)
Comment (JamieDraperUK) [reply build.gradle:90]: Can't see why not it's just a test implementation.

**Review by Nimphal — APPROVED** [review]
LGTM

**Pull request merged**